source/core/TypeAndDataManager.cpp

```diff
@@ -1180,20 +1188,21 @@ TypeRef WrappedType::GetSubElementAddressFromPath(SubString* name, void *start,
 //------------------------------------------------------------
 // ElementType
 //------------------------------------------------------------
-ElementType* ElementType::New(TypeManagerRef typeManager, SubString* name, TypeRef wrappedType, UsageTypeEnum usageType, Int32 offset)
+ElementType* ElementType::New(TypeManagerRef typeManager, SubString* name, TypeRef wrappedType, UsageTypeEnum usageType, Int32 offset, bool isDataItem)
 {
-    ElementType* type = TADM_NEW_PLACEMENT_DYNAMIC(ElementType, name)(typeManager, name, wrappedType, usageType, offset);
-
+    ElementType* type = TADM_NEW_PLACEMENT_DYNAMIC(ElementType, name)(typeManager, name, wrappedType, usageType, offset, isDataItem);
+    if (isDataItem)
```

Comment:
> Why do we early return from here if it is data item?
Because we don't want the types unique'd for dataItems, since we need to track the updateNeeded flag independently.

> In VIA generated from DFIR , for aggregate data types such as clusters, only the variable itself will be marked "d" (data item) and not every field in the aggregate data type.
This is fine, because when you write to a terminal in G, you always update the entire thing, which will generate a Copy instruction to the entire cluster.